connectors/protocol/connectors.py

```diff
@@ -393,29 +409,50 @@ async def fail(self, error, ingestion_stats=None, connector_metadata=None):
             else:
                 message = str(error)
             await self._terminate(
-                JobStatus.ERROR, message, ingestion_stats, connector_metadata
+                JobStatus.ERROR,
+                message,
+                ingestion_stats,
+                connector_metadata,
+                checkpoint,
             )
         except Exception as e:
             self._wrap_errors("terminate as failed", e)
 
-    async def cancel(self, ingestion_stats=None, connector_metadata=None):
+    async def cancel(
+        self, ingestion_stats=None, connector_metadata=None, checkpoint=None
+    ):
         try:
             await self._terminate(
-                JobStatus.CANCELED, None, ingestion_stats, connector_metadata
+                JobStatus.CANCELED,
+                None,
+                ingestion_stats,
+                connector_metadata,
+                checkpoint,
             )
         except Exception as e:
             self._wrap_errors("terminate as canceled", e)
 
-    async def suspend(self, ingestion_stats=None, connector_metadata=None):
+    async def suspend(
+        self, ingestion_stats=None, connector_metadata=None, checkpoint=None
+    ):
         try:
             await self._terminate(
-                JobStatus.SUSPENDED, None, ingestion_stats, connector_metadata
+                JobStatus.SUSPENDED,
+                None,
+                ingestion_stats,
+                connector_metadata,
+                checkpoint,
             )
         except Exception as e:
             self._wrap_errors("terminate as suspended", e)
 
     async def _terminate(
-        self, status, error=None, ingestion_stats=None, connector_metadata=None
+        self,
+        status,
+        error=None,
+        ingestion_stats=None,
+        connector_metadata=None,
+        checkpoint=None,
```

Comment:
Why do we need to update checkpoint in the `_terminate` operation? 

We are updating job metadata (docs ingested etc) but I don't see any changes w.r.t. to done/suspend/error functions in the `sync_job_runner` (those methods don't care about the current checkpoint object)